### DISCUSSION TITLE
Import path as song field

### DISCUSSION THREAD

**Discussion by AndreKR** [navidrome/navidrome#2304] [Ideas]
I have my music organized by mood/occasion. I couldn't care less about an artist's name, let alone an album name. (Seriously, the notion of an "album" seems bizarre to me.) I already have the files sorted into those mood/occasion categories. Now I want to use Navidrome but unfortunately currently the source path seems to be completely lost when importing files into Navidrome.

Would it be possible to add the source path like a tag so I can show it in the UI and search in it?

**Comment by deluan** [comment]
Navidrome does not support browse by folder, so unfortunately it is not really suitable for your use case. Anyway, you can show the filepaths as a column in the Songs list:

https://user-images.githubusercontent.com/331353/230752450-b50014d8-7bf2-4f50-9d0b-b8117dd0d05f.mov

**Reply by AndreKR** [reply]
What's the name of the column? I don't see any column that would fit the purpose.

![image](https://user-images.githubusercontent.com/1188538/230754098-2895bb3d-95b8-4a90-ab1d-fb9355a9fbec.png)

**Comment by deluan** [comment]
Ops, sorry. It is not released yet, It will be in the next version. It is only available in dev builds. We will probably have another release soon, but if you want to try it, you can get the latest dev build from docker (`develop` tag) or from the GitHub project's page, by clicking here:
<img width="458" alt="Screenshot 2023-04-09 at 1 46 22 AM" src="https://user-images.githubusercontent.com/331353/230756637-101a977d-8b72-431b-9ed4-1b011e7d7367.png">

**Reply by AndreKR** [reply]
Do you use feature branches and `develop` is basically releasable or will I get a potpourri of half-finished features when I use `develop`? 😄

**Reply by deluan** [reply]
I try to do trunk-based development as much as possible, avoiding creating feature branches. Because of that, `develop` is _usually_ in a releasable state. If there's work in progress in `develop`, it should not break anything or it would be disabled behind a [Dev](https://github.com/navidrome/navidrome/blob/master/conf/configuration.go#L86) feature flag. But it could also be broken, so proceed with caution ;) 

Having said that, I personally use `develop` in my main personal instance of Navidrome, updated daily, so bugs/broken features don't stay that way for long. And I never had a situation where a broken feature caused any irreversible damage.

**Reply by AndreKR** [reply]
> I try to do trunk-based development as much as possible, avoiding creating feature branches. Because of that, develop is usually in a releasable state.

Shouldn't it be the other way around? If you *don't* have feature branches, half-finished features end up in `develop`.

Anyway, I found a better solution for myself: I wrote a script that puts a file `<name of folder>.m3u` into each folder, simply listing the files in the folder one filename per line. Then I enabled `ND_AUTOIMPORTPLAYLISTS` and now all my folders show up as playlists.

**Reply by deluan** [reply]
> Shouldn't it be the other way around? If you don't have feature branches, half-finished features end up in develop.

Yes and no. Yes, there will be more unfinished features in master. No, they won't appear to the user unless they are in a workable state (or the user enabled the feature flag). It also forces me to always keep `develop` in a sane state, instead of mindless pushing semi-baked code to a branch. 

The [premise](https://trunkbaseddevelopment.com/5-min-overview/) is: _Trunk-Based Development will always be release ready._


> Anyway, I found a better solution for myself: I wrote a script that puts a file <name of folder>.m3u into each folder, simply listing the files in the folder one filename per line. Then I enabled ND_AUTOIMPORTPLAYLISTS and now all my folders show up as playlists.

Ah, nice. If you have just a few folders like that, you could create one [Smart Playlist](https://github.com/navidrome/navidrome/issues/1417#issuecomment-1204111835) for each one.  This way you don't have to recreate the M3U every time you add/remove a file.
